### PR TITLE
uf-6347: Updated customer details with paging and new parameters.

### PR DESCRIPTION
**Pull request by mha109**

### CODE REVIEW

**Review by cheezi0747 — APPROVED** [review]

File: src/integration-test/resources/ReadDetails/__files/test1_readDetailsByPartyIdAndEngagementOrgId/request.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "partyId": [
```
Comment (foobar-1337) [inline src/integration-test/resources/ReadDetails/__files/test1_readDetailsByPartyIdAndEngagementOrgId/request.json:2]: Use the same indentation as the rest of the json-files?

(comment applicable on other json files as well)

File: src/integration-test/java/apptest/ReadDetailsIT.java [hunk 2]
```diff
@@ -87,7 +69,7 @@ void test4_readDetailsByCustomerEngagementOrgId() throws Exception {
 	}
 
 	@Test
-	void test5_readDetailsByCustomerEngagementOrgIdNotFound() throws Exception {
+	void test4_readDetailsByCustomerEngagementOrgIdNotFound() throws Exception {
```
Comment (josc04) [inline src/integration-test/java/apptest/ReadDetailsIT.java:72]: This test and test2_readDetailsByPartyIdNotFound looks very similar, do they not? Remove one of them?

File: src/integration-test/java/apptest/ReadDetailsIT.java
Comment (josc04) [inline src/integration-test/java/apptest/ReadDetailsIT.java:1]: No test with multiple partyIds?

Comment (josc04) [inline src/integration-test/java/apptest/ReadDetailsIT.java:1]: No test with multiple partyIds?

File: src/integration-test/java/apptest/ReadDetailsIT.java [hunk 3]
```diff
@@ -104,7 +86,7 @@ void test5_readDetailsByCustomerEngagementOrgIdNotFound() throws Exception {
 	}
 
 	@Test
-	void test6_readDetailsByCustomerEngagementOrgIdWithFromDate() throws Exception {
+	void test5_readDetailsByCustomerEngagementOrgIdWithFromDate() throws Exception {
 		final var result = setupCall()
```
Comment (foobar-1337) [inline src/integration-test/java/apptest/ReadDetailsIT.java:90]: Why are you returning a body in this method?

You have: 
`.withExpectedResponse("response.json")`

Doesn't this cover all response verifications?

File: src/main/java/se/sundsvall/customer/api/validation/impl/ValidCustomerDetailsRequestConstraintValidator.java [hunk 4]
```diff
@@ -12,6 +12,6 @@ public class ValidCustomerDetailsRequestConstraintValidator implements Constrain
 
 	@Override
 	public boolean isValid(final CustomerDetailsRequest request, final ConstraintValidatorContext context) {
-		return !(isEmpty(request.getPartyId()) && isBlank(request.getCustomerEngagementOrgId()));
+		return !(isBlank(request.getCustomerEngagementOrgId()));
```
Comment (josc04) [inline src/main/java/se/sundsvall/customer/api/validation/impl/ValidCustomerDetailsRequestConstraintValidator.java:15]: Isn't this validator obsolete now that the `@ValidOrganizationNumber(nullable = true)` has been changed to not allow nulls (`@ValidOrganizationNumber`)?

File: src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java [hunk 5]
```diff
@@ -42,15 +44,21 @@ void hasValidBuilderMethods() {
 			.withPartyId(List.of(partyId))
 			.withCustomerEngagementOrgId(customerEngagementOrgId)
 			.withFromDateTime(fromDateTime);
+		request.setSortBy(List.of("sort1", "sort2"));
```
Comment (josc04) [inline src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java:48]: Set page and limit as well and verify that they are not the "default values"?

File: src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java [hunk 6]
```diff
@@ -42,15 +44,21 @@ void hasValidBuilderMethods() {
...
+		assertThat(request.getSortBy()).containsExactly("sort1", "sort2");
+		assertThat(request.getSortDirection()).hasToString(Direction.ASC.toString());
 	}
 
 	@Test
 	void hasNoDirtOnCreatedBean() {
-		assertThat(new CustomerDetailsRequest()).hasAllNullFieldsOrProperties();
+		var customerDetailsRequest = new CustomerDetailsRequest();
+		assertThat(customerDetailsRequest).hasAllNullFieldsOrPropertiesExcept("page", "limit", "sortDirection");
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java:63]: Assert default values for page, limit and sortDirection?

File: src/integration-test/resources/ReadDetails/__files/test1_readDetailsByPartyIdAndEngagementOrgId/request.json (hunk 1 above)
Comment (mha109) [reply src/integration-test/resources/ReadDetails/__files/test1_readDetailsByPartyIdAndEngagementOrgId/request.json:2]: It was a mixed bag when I came here, but I'll fix.

File: src/integration-test/java/apptest/ReadDetailsIT.java (hunk 3 above)
Comment (mha109) [reply src/integration-test/java/apptest/ReadDetailsIT.java:90]: Good question, the tests were already written and I didn't reflect on the additional asserts, I'll remove them

File: src/main/java/se/sundsvall/customer/api/validation/impl/ValidCustomerDetailsRequestConstraintValidator.java (hunk 4 above)
Comment (mha109) [reply src/main/java/se/sundsvall/customer/api/validation/impl/ValidCustomerDetailsRequestConstraintValidator.java:15]: Yes.

File: src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java (hunk 5 above)
Comment (mha109) [reply src/test/java/se/sundsvall/customer/api/model/CustomerDetailsRequestTest.java:48]: Will fix

File: src/integration-test/java/apptest/ReadDetailsIT.java
Comment (mha109) [reply src/integration-test/java/apptest/ReadDetailsIT.java:1]: Duplicate? ;-)

**Review by josc04 — DISMISSED** [review]

**Review by josc04 — APPROVED** [review]

**Pull request merged**